File: bio/biobase/eset/wrapper.R
# __author__ = "Leandro C. Hermida"
# __email__ = "hermidalc@pitt.edu"
# __license__ = "MIT"

suppressPackageStartupMessages({
    library(Biobase)
    library(data.table)
})

# Sink the stderr and stdout to the snakemake log file
# https://stackoverflow.com/a/48173272
log <- file(snakemake@log[[1]], open = "wt")
sink(log)
sink(log, type = "message")

adata <- fread(snakemake@input[["assay"]], header = TRUE)
pdata <- read.delim(
    snakemake@input[["pheno"]],
    sep = "\t", header = TRUE,
    row.names = snakemake@params[["samples"]]
)
fdata <- read.delim(
    snakemake@input[["annot"]],
    sep = "\t", header = TRUE, row.names = 1
)

eset <- ExpressionSet(
    assayData = as.matrix(adata, rownames = 1),
    phenoData = AnnotatedDataFrame(pdata),
    featureData = AnnotatedDataFrame(fdata),
)

saveRDS(eset, file = snakemake@output[[1]])

# Proper syntax to close the connection for the log file but could be optional
# for Snakemake wrapper
sink(type = "message")
sink()

File: utils/pigz/decompress/wrapper.py
__author__ = "Leandro C. Hermida"
__email__ = "hermidalc@pitt.edu"
__license__ = "BSD 3-Clause"

from snakemake.shell import shell

log = snakemake.log_fmt_shell(stdout=False, stderr=True)

file_type = (
    "--zip"
    if snakemake.input[0].endswith((".zip", ".ZIP"))
    else "--zlib"
    if snakemake.input[0].endswith((".zz", ".ZZ"))
    else ""
)

shell(
    "pigz"
    " --processes {snakemake.threads}"
    " {file_type}"
    " --decompress"
    " --stdout {snakemake.input[0]} > {snakemake.output[0]}"
    " {log}"
)
